utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/universal/UniversalSchemaBasedSerdeProvider.scala

```diff
@@ -20,15 +22,18 @@ object UniversalSchemaBasedSerdeProvider {
 
   private def createSchemaIdFromMessageExtractor(schemaRegistryClient: SchemaRegistryClient): ChainedSchemaIdFromMessageExtractor = {
     val isConfluent = schemaRegistryClient.isInstanceOf[ConfluentSchemaRegistryClient]
-    createSchemaIdFromMessageExtractor(isConfluent)
+    val isAzure = schemaRegistryClient.isInstanceOf[AzureSchemaRegistryClient]
+    createSchemaIdFromMessageExtractor(isConfluent, isAzure)
   }
 
   // SchemaId can be obtain in several ways. Precedent:
   // * from nu kafka headers - it is our own, Nussknacker headers standard format: key.schemaId and value.schemaId headers
+  // * from azure header - content-type: avro/binary+schemaId (only value schema ids are supported)
   // * from payload serialized in 'Confluent way' ([magicbyte][schemaid][payload])
   // * (fallback) from source editor version param - this is just an assumption we make (when processing no-schemed-data, everything can happen)
-  private[schemaregistry] def createSchemaIdFromMessageExtractor(isConfluent: Boolean): ChainedSchemaIdFromMessageExtractor = {
+  private[schemaregistry] def createSchemaIdFromMessageExtractor(isConfluent: Boolean, isAzure: Boolean): ChainedSchemaIdFromMessageExtractor = {
```

Comment:
WDYT about one parameter -`providerType` which is enum or something, instead of multiple Boolean flags?

Reply:
I like current approach because it handles 3rd option that we haven't got yet - not confluent format, not azure format - only our headers. For example we could easily add support of json schemas (or some other) kept on file system (we already have one question from our clients to handle such situation). Current approach will cause changes In only needed places - no need to think how to handle it - simple it would work - no confluent logic, no azure logic will happen, only our headers. But I understand why you don't like that. If you are very convicted to do such a change, I can do it, because it is rather small thing to change anytime

Reply:
No it's ok. I see a value in what you said